crates/relayer/src/chain/cosmos.rs

```diff
@@ -388,6 +388,10 @@ impl CosmosSdkChain {
     pub fn unbonding_period(&self) -> Result<Duration, Error> {
         crate::time!("unbonding_period");
 
+        if let Some(unbonding_period) = self.config.unbonding_period {
+            return Ok(unbonding_period);
+        }
+
```

Comment:
Should we consider doing this only if the call below shows status `Unimplemented`? And use a non-zero default for unbonding_period.

Also there is a call to `chain.historical_entries()` in `do_health_check()` where we should maybe issue a warning instead of returning error so further checks are done. Or just move this check to be last.

Reply:
> Should we consider doing this only if the call below shows status Unimplemented? And use a non-zero default for unbonding_period.

I think we should always use the setting's value if it's present, otherwise we may confuse operators when the clients they create etc. don't have the unbonding period they've specified. There is an always chance that they may set it even for a non-ICS consumer chain and that they may set the wrong value, but that's the case for pretty much every setting we have already.

> Also there is a call to chain.historical_entries() in do_health_check() where we should maybe issue a warning instead of returning error so further checks are done. Or just move this check to be last.

Agreed.

Reply:
> ...but that's the case for pretty much every setting we have already.

afaik is for parameters that we cannot query. In this case `unbonding_period` could be set to `X` and a query to chain could give `Y`.  Using a bad `X` could result in client errors later during relaying.
Maybe we could have some warning in the health check at least?